### PR TITLE
Fix UBL mesh editor on delta

### PR DESCRIPTION
**Pull request by tcm0116**
### Description

This PR fixes the UBL mesh editor to be compatible with delta machines.

### CODE REVIEW

**Comment by Roxy-3D** [comment]
Thank You!    I wasn't aware the mesh editor was broken for Deltas.    What was it doing wrong?

**Comment by tcm0116** [comment]
@Roxy-3D - it was trying to move to positions on the grid that are outside the bounds of the machine. It was also using unsegmented movement commands, which cause the nozzle to dig into the bed during long movements.

File: Marlin/src/lcd/menu/menu_ubl.cpp [hunk 1]
```diff
@@ -461,22 +473,36 @@ void _lcd_ubl_output_map_lcd() {
...
+      // Prevent underrun/overrun of plot numbers
+      x_plot = constrain(x_plot, GRID_MAX_POINTS_X - (GRID_MAX_POINTS_X + 1), GRID_MAX_POINTS_X + 1);
+      y_plot = constrain(y_plot, GRID_MAX_POINTS_Y - (GRID_MAX_POINTS_Y + 1), GRID_MAX_POINTS_Y + 1);
 
-  // Prevent underrun/overrun of plot numbers
-  x_plot = constrain(x_plot, GRID_MAX_POINTS_X - (GRID_MAX_POINTS_X + 1), GRID_MAX_POINTS_X + 1);
-  y_plot = constrain(y_plot, GRID_MAX_POINTS_Y - (GRID_MAX_POINTS_Y + 1), GRID_MAX_POINTS_Y + 1);
+  #if IS_KINEMATIC
+      if (!position_is_reachable(pgm_read_float(&ubl._mesh_index_to_xpos[x_plot]),
+                                 pgm_read_float(&ubl._mesh_index_to_xpos[y_plot]))) {
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_ubl.cpp]: `_mesh_index_to_ypos` … now patched!

File: Marlin/src/lcd/menu/menu_ubl.cpp [hunk 2]
```diff
@@ -461,22 +473,37 @@ void _lcd_ubl_output_map_lcd() {
...
+  #endif
+
+  do {
+    // Encoder to the right (++)
+    if (x_plot >= GRID_MAX_POINTS_X) { x_plot = 0; y_plot++; }
+    if (y_plot >= GRID_MAX_POINTS_Y) y_plot = 0;
+
+    // Encoder to the left (--)
+    if (x_plot <= GRID_MAX_POINTS_X - (GRID_MAX_POINTS_X + 1)) { x_plot = GRID_MAX_POINTS_X - 1; y_plot--; }
+    if (y_plot <= GRID_MAX_POINTS_Y - (GRID_MAX_POINTS_Y + 1)) y_plot = GRID_MAX_POINTS_Y - 1;
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_ubl.cpp]: This actually just reduces to `if (y_plot < 0) y_plot = GRID_MAX_POINTS_Y - 1;` and the same with `x_plot`. Another note below.

File: Marlin/src/lcd/menu/menu_ubl.cpp [hunk 3]
```diff
@@ -461,22 +473,37 @@ void _lcd_ubl_output_map_lcd() {
...
 
-  // Encoder to the left (--)
-  if (x_plot <= GRID_MAX_POINTS_X - (GRID_MAX_POINTS_X + 1)) { x_plot = GRID_MAX_POINTS_X - 1; y_plot--; }
-  if (y_plot <= GRID_MAX_POINTS_Y - (GRID_MAX_POINTS_Y + 1)) y_plot = GRID_MAX_POINTS_Y - 1;
+    // Prevent underrun/overrun of plot numbers
+    x_plot = constrain(x_plot, GRID_MAX_POINTS_X - (GRID_MAX_POINTS_X + 1), GRID_MAX_POINTS_X + 1);
+    y_plot = constrain(y_plot, GRID_MAX_POINTS_Y - (GRID_MAX_POINTS_Y + 1), GRID_MAX_POINTS_Y + 1);
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_ubl.cpp]: This reduces to `y_plot = constrain(y_plot, -1, GRID_MAX_POINTS_Y + 1);` (`x_plot` too). Is it okay if `x_plot` and/or `y_plot` are -1?

I guess this is what was in the original code. Hmm!

**Comment by thinkyhead** [comment]
The last commit reduces the lower bounds test and comments out the redundant constraint between `-1` and `GRID_MAX_POINTS_X + 1`, since that constraint (a tighter constraint) is applied just above.

**Comment by tcm0116** [comment]
@thinkyhead - your changes look good to me

**Pull request merged**